### DISCUSSION TITLE
Change multiple fields with `setFieldValue`?

### DISCUSSION THREAD

**Discussion by 1cgonza** [directus/directus#16943] [Archive]
I am trying to reset the form with a custom interface, but using more than one `setFieldValue` does not seem to work:

This works:
```js
emit('setFieldValue', { field: 'titulo', value: null });
```

This is not working for both, but just changes the last one:
```js
emit('setFieldValue', { field: 'titulo', value: null });
emit('setFieldValue', { field: 'fuente', value: null });
```

Is there a way to change more than one field value?

**Comment by nickreese** [comment]
Working on this exact scenario. Will keep you posted on what I find.

**Reply by nickreese** [reply]
https://github.com/directus/directus/pull/11620#issuecomment-1141053792

**Reply by nickreese** [reply]
```
emit("setFieldValue", { field: "address", value: "test updating" });
emit("setFieldValue", { field: "other", value: "test updating" });
```
Works just fine for me.

**Reply by 1cgonza** [reply]
Thank you @nickreese for checking this. Weird it is not working for me, wonder what is the difference. Do you mind sharing a bit more info on your side?
I am using Directus 9.22.1.
Here is a simple example of what I am trying:

```js
<template>
  <button @click="borrar">Borrar Datos</button>
</template>

<script>
import { defineComponent } from 'vue';

export default defineComponent({
  emits: ['input'],
  props: {
    collection: {
      type: String,
      default: null,
    },
  },

  setup({}, { emit }) {
    return { borrar };

    function borrar() {
      emit('setFieldValue', { field: 'titulo', value: null });
      emit('setFieldValue', { field: 'fuente', value: null });
    }
  },
});
</script>
```

**Reply by nickreese** [reply]
OK, now that my full component is written, it isn't working for me either. Just the last one is updated.

**Reply by 1cgonza** [reply]
@nickreese did you figure out how to do it?

**Reply by kesslerdev** [reply]
It's definitely a bug.
I experienced this bug with the extension [directus-extension-computed-interface](https://github.com/rezo-labs/directus-extension-computed-interface) when a field is used in more than one expression only the last one changed, the "ugly" workaround in my case is to use setTimeout.

file => https://github.com/rezo-labs/directus-extension-computed-interface/blob/main/src/interface.vue#L90

```javascript
setTimeout(() => {
    emit('input', newValue);
}, 1);
```

**Reply by lambrusco** [reply]
Hey, I faced the same issue, is there already one issue open for this?

**Reply by LeonManolo** [reply]
Im facing the same issue

**Comment by kgkg** [comment]
I'm not sure if it's bug or intended behaviour, but what I found out is when you use `nextTick` you can update more fields.

Example with 2 fields:
```ts
  emit("setFieldValue", {field: "title", value: some_value});
  nextTick(() => emit("setFieldValue", {field: "description", value: some_value}));
```

It gets wilder the more fields you want to change :)
```ts
  emit("setFieldValue", {field: "title", value: some_value});
  nextTick(() => {
    emit("setFieldValue", {field: "description", value: some_value});
    nextTick(() => emit("setFieldValue", {field: "image_url", value: some_value})
  });
```

Event to change many fields at once would be very handy :)
